plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1480,3 +1480,49 @@ func (c *VolumeNodeChecker) predicate(pod *v1.Pod, meta algorithm.PredicateMetad
 	}
 	return true, nil, nil
 }
+
+type NoDeletedPVCChecker struct {
+	pvcInfo PersistentVolumeClaimInfo
+}
+
+// NewNoDeletedPVCsPredicate evaluates if a pod can fit due to the PVC it uses.
+func NewNoDeletedPVCsPredicate(pvcInfo PersistentVolumeClaimInfo) algorithm.FitPredicate {
+	c := &NoDeletedPVCChecker{
+		pvcInfo: pvcInfo,
+	}
+	return c.predicate
+}
+
+func (c *NoDeletedPVCChecker) predicate(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, error) {
```

Comment:
We don't need to run this for every node. Since scheduler doesn't have a better place to put this code, please add a comment with a TODO to mention that it should be run only once per pod.